engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetViewportSizeOf.java

```diff
@@ -27,35 +27,33 @@
  */
 public final class SetViewportSizeOf implements FBOManagerSubscriber, StateChange {
     private static final String DEFAULT_FBO = "sceneOpaque";
-    private static SetViewportSizeOf defaultInstance;
+    private static SetViewportSizeOf defaultInstance = new SetViewportSizeOf(DEFAULT_FBO);
+    private static FrameBuffersManager frameBuffersManager;
     private SetViewportSizeOfTask task;
 
     private FBO fbo;
     private String fboName;
-    private FrameBuffersManager frameBuffersManager;
 
-    public SetViewportSizeOf(String fboName, FrameBuffersManager frameBuffersManager) {
+    public SetViewportSizeOf(String fboName) {
         this.fboName = fboName;
-        this.frameBuffersManager = frameBuffersManager;
-        fbo = frameBuffersManager.getFBO(fboName);
+        if (!fboName.equals(DEFAULT_FBO)) {
+            fbo = frameBuffersManager.getFBO(fboName);
+        }
```

Comment:
(...some time later...) 

I'm pretty convinced that you can remove the whole IF block, in line with the comment regarding the toString() method. It's important that the instance of this state change holds the fboName. But the fbo itself is only needed briefly when the task is generated or when the FBM signals a change through the update() method. So, I don't think we need to store the fbo permanently nor keep it up to date.